lib/ui/questions/survey_questions_screen.dart

```diff
@@ -0,0 +1,192 @@
+                children: [
+                  _buildCloseSurveyButton(),
+                  _buildQuestionIndicator(visibleIndex, totalQuestions),
+                  const SizedBox(height: space8),
+                  _buildQuestionsView(surveyDetails.questions),
+                  _buildNextQuestionButton(visibleIndex, totalQuestions),
+                  const SizedBox(height: space57)
+                ],
+              ),
+            ),
+          ),
+        ),
+      );
+
+  Widget _buildCloseSurveyButton() => Padding(
+        padding:
+            const EdgeInsets.symmetric(horizontal: space16, vertical: space20),
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.end,
+          children: [
+            IconButton(
+              key: SurveyQuestionsWidgetId.closeSurveyButton,
+              icon: SvgPicture.asset(Assets.images.icClose.path),
+              onPressed: () => context.pop(), // TODO Add confirmation dialog
+            )
+          ],
+        ),
+      );
+
+  Widget _buildQuestionIndicator(int visibleIndex, int totalQuestion) =>
+      Padding(
+        padding: const EdgeInsets.symmetric(horizontal: space20),
+        child: Text(
+          '${visibleIndex + 1}/$totalQuestion',
+          key: SurveyQuestionsWidgetId.questionIndicator,
+          style: TextStyle(
+              color: Colors.white.withOpacity(0.5),
+              fontSize: 15,
+              fontWeight: FontWeight.bold),
+        ),
+      );
+
+  Widget _buildQuestionsView(List<SurveyQuestionUiModel> questions) => Consumer(
+        builder: (_, ref, __) {
+          return Expanded(
+            child: SurveyQuestionView(
+              questions: questions,
+              onPageChanged: (index) => {
+                ref
+                    .read(surveyQuestionsViewModelProvider.notifier)
+                    .setVisibleSurveyIndex(index)
+              },
+            ),
+          );
+        },
+      );
+
+  Widget _buildNextQuestionButton(int visibleIndex, int totalQuestion) =>
+      Padding(
+        padding: const EdgeInsets.symmetric(horizontal: space20),
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.end,
+          children: [
+            // TODO Show the Submit survey button when the last question is shown
+            Visibility(
+              visible: visibleIndex < totalQuestion - 1,
+              child: WhiteRightArrowButton(
+                key: SurveyQuestionsWidgetId.nextQuestionButton,
+                onPressed: () {
+                  ref
+                      .read(surveyQuestionsViewModelProvider.notifier)
+                      .nextQuestion();
+                },
+              ),
+            ),
+          ],
+        ),
+      );
+
+  // TODO: Extract to common widget for reuse
+  Widget _buildLoadingIndicator() {
+    return Center(
+      child: Visibility(
+        maintainSize: true,
+        maintainAnimation: true,
+        maintainState: true,
+        visible: true,
```

Comment:
@hoangnguyen92dn why do we need `Visibility` here? I think we can remove since it's redundant 🤔

Reply:
@manh-t That's right. My intention is to create a common Widget for the loading indicator and handle show/hide logic. I could remove it but currently please leave it there since we will do refactor soon.